Create a Task for 'Creating your own secrets' #16006 (#16176)
kubernetes/website: content/en/docs/tasks/manage-secrets/creating-your-own-secrets.md

---
content_template: templates/concept
title: Creating your own Secrets
---

{{% capture overview %}}

> This page describes how users can create their own secrets. See [Secrets](/docs/concepts/configuration/secret/) for more information.

Reviewer: ```suggestion
This page shows you how to create new Secrets. See [Secrets](/docs/concepts/configuration/secret/) for more information.
```

```

You can check that the secret was created like this:

```shell
$ kubectl get secrets
NAME                             TYPE                                  DATA      AGE
db-user-pass-96mffmfh4k          Opaque                                2         51s

$ kubectl describe secrets/db-user-pass-96mffmfh4k
Name:            db-user-pass
Namespace:       default
Labels:          <none>
Annotations:     <none>

Type:            Opaque

Data
====
password.txt:    12 bytes
username.txt:    5 bytes
```

For example, to generate a Secret from literals `username=admin` and `password=secret`,
you can specify the secret generator in `kustomization.yaml` as
```shell
# Create a kustomization.yaml file with SecretGenerator
$ cat <<EOF >./kustomization.yaml
secretGenerator:
- name: db-user-pass
  literals:
  - username=admin
  - password=secret
EOF
```
Apply the kustomization directory to create the Secret object.
```shell
$ kubectl apply -k .
secret/db-user-pass-dddghtt9b5 created
```
{{< note >}}
The generated Secrets name has a suffix appended by hashing the contents. This ensures that a new
Secret is generated each time the contents is modified.
{{< /note >}}

> 

Author: ```suggestion
{{% capture whatsnext %}}

* Read [Create a Pod that has access to the secret data through a Volume](/docs/tasks/inject-data-application/distribute-credentials-secure/#create-a-pod-that-has-access-to-the-secret-data-through-a-volume)
```

> {{% /capture %}}

Reviewer: Does this page warrant a What's Next section?

Author: @sftim I have added What's Next section above. What do you think about having "Create a Pod that has access to the secret data through a Volume" as next reading suggestion?

Reviewer: Sounds good to me. Also consider https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/

Reviewer: Actually, I wonder whether it makes sense to snip that content out of the Concept page and merge it into https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/

Otherwise, there will be two Tasks that cover similar ground. I had not realised that when I logged the original issue.

Author: Do you mean taking out https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets (Using Secrets) section from concept -> Secrets page and merge into https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/ ?

There is also duplication of https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#create-a-secret with our Tasks page "Creating Your Own Secrets" (https://deploy-preview-16176--kubernetes-io-master-staging.netlify.com/docs/tasks/manage-secrets/creating-your-own-secrets/ ).

Reviewer: There's different ways of tidying this up. What I've realized here is that my original recommendations don't necessarily work well. Sorry about that.
